app/views/Project/Tagging/Sources/SourcesFilter/EntryFilter/FrameworkFilterItem/index.tsx

```diff
@@ -0,0 +1,389 @@
+const filterClientIdSelector = (d: KeyLabelEntity) => d.clientId;
+const filterLabelSelector = (d: KeyLabelEntity | KeyLabel) => d.label;
+
+type PartialFrameworkFilterValue = NonNullable<PartialEntriesFilterDataType['filterableData']>[number];
+
+interface Props<K extends number> {
+    name: K;
+    title: string;
+    filter: FrameworkFilterType;
+    // FIXME: the filter's value should depend on the filter type as well
+    value: PartialFrameworkFilterValue | undefined;
+    onChange: (value: SetValueArg<PartialFrameworkFilterValue>, name: K) => void;
+    projectId: string;
+    allFiltersVisible: boolean;
+    className?: string;
+    optionsDisabled?: boolean;
+    disabled?: boolean;
+}
+
+function FrameworkFilterItem<K extends number>(props: Props<K>) {
+    const {
+        name,
+        title,
+        value,
+        filter,
+        projectId,
+        className,
+        onChange,
+        allFiltersVisible,
+        optionsDisabled,
+        disabled,
+    } = props;
+
+    const [
+        geoAreaOptions,
+        setGeoAreaOptions,
+    ] = useState<GeoArea[] | undefined | null>(undefined);
+
+    const defaultOptionVal = useCallback(
+        (): PartialFrameworkFilterValue => ({
+            filterKey: filter.key,
+        }),
+        [filter],
+    );
+    const onFieldChange = useFormObject(name, onChange, defaultOptionVal);
+
+    // FIXME: move this logic to it's own component
+    const handleDateRangeChange = useCallback(
+        (dateRangeValue: { startDate: string, endDate: string } | undefined) => {
+            if (isNotDefined(dateRangeValue)) {
+                onChange({ filterKey: filter.key }, name);
+            } else {
+                onChange(
+                    (oldVal) => ({
+                        ...oldVal,
+                        filterKey: filter.key,
+                        valueGte: convertDateToIsoDateTime(dateRangeValue.startDate),
+                        valueLte: convertDateToIsoDateTime(dateRangeValue.endDate),
+                    }),
+                    name,
+                );
+            }
+        },
+        [name, onChange, filter.key],
+    );
+
+    const handleTimeRangeChange = useCallback(
+        (timeRangeValue: { startTime: string, endTime: string } | undefined) => {
+            if (isNotDefined(timeRangeValue)) {
+                onChange({ filterKey: filter.key }, name);
+            } else {
+                onChange(
+                    (oldVal) => ({
+                        ...oldVal,
+                        filterKey: filter.key,
+                        valueGte: timeRangeValue.startTime,
+                        valueLte: timeRangeValue.endTime,
+                    }),
+                    name,
+                );
+            }
+        },
+        [name, onChange, filter.key],
+    );
+
+    switch (filter.widgetType) {
+        case 'DATE': {
+            return (
+                <DateInput
```

Comment:
We may need to use ranged input for date, time and number inputs
@AdityaKhatri @thenav56 any thoughts?

Reply:
Yes.